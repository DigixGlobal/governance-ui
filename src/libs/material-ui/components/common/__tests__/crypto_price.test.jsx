import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CryptoPrice from '../crypto_price';

jest.mock('react-redux', () => ({
  connect: () => component => component,
}));

const cryptoPrices = {
  available: ['Test'],
  prices: {
    GBP: 10,
    selected: 'GBP',
  },
  selected: 'GBP',
};

describe('<CryptoPrice />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <CryptoPrice
        ammount={400}
        cryptoPrices={cryptoPrices}
        getPrices={jest.fn}
        setSelectedCurrency={jest.fn}
        symbol="GBP"
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
