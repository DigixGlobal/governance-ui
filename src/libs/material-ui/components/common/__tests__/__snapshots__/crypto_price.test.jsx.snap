// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CryptoPrice /> renders correctly 1`] = `
<Label
  ammount={400}
  basic={true}
  getPrices={[Function]}
  pointing={true}
  setSelectedCurrency={[Function]}
>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    icon="dropdown"
    minCharacters={1}
    noResultsMessage="No results found."
    onChange={[Function]}
    openOnFocus={true}
    options={
      Array [
        Object {
          "text": "Test",
          "value": "Test",
        },
      ]
    }
    renderLabel={[Function]}
    scrolling={true}
    searchInput="text"
    selectOnBlur={true}
    selectOnNavigation={true}
    text="About 0 GBP"
    value="GBP"
  />
</Label>
`;
